src/amazonqFeatureDev/errors.ts

```diff
@@ -64,6 +70,12 @@ export class ContentLengthError extends ToolkitError {
     }
 }
 
+export class ApiError extends ToolkitError {
+    constructor(message: string, api: string, errorCode: number) {
+        super(message, { code: `${api}-${errorCode}` })
```

Comment:
I would prefer here to have : `${api}-${errorName}-${errorCode}`
In that way it add more readability for the people not familiar with error codes